core/unit_test/cuda/TestCuda_Graph.cpp

```diff
@@ -16,3 +16,67 @@
 
 #include <TestCuda_Category.hpp>
 #include <TestGraph.hpp>
+
+namespace Test {
+
+/**
+ * @test This test ensures that the underlying @c CUDA objects can be retrieved
+ *       from a @c Kokkos::Graph.
+ *
+ * It also ensures that these objects can be modified "externally", thereby
+ * testing "interoperability".
+ */
+TEST(TEST_CATEGORY, cuda_graph_instantiate_and_debug_dot_print) {
+  //! Build a rank-1 zero-initialized view.
+  using view_t   = Kokkos::View<int, TEST_EXECSPACE>;
+  using view_h_t = Kokkos::View<int, Kokkos::HostSpace>;
+  view_t data("witness");
+
+  //! Build a graph, whose sole node will apply @ref AddOne on @ref data.
+  auto graph =
+      Kokkos::Experimental::create_graph<TEST_EXECSPACE>([&](auto root) {
+        root.then_parallel_for(
+            1, SetViewToValueFunctor<TEST_EXECSPACE, int>{data, 1});
+      });
+
+  //! Retrieve the @c Kokkos::Impl::Graph.
+  auto root_node_ref = Kokkos::Impl::GraphAccess::create_root_ref(graph);
+  auto graph_ptr_impl =
+      Kokkos::Impl::GraphAccess::get_graph_weak_ptr(root_node_ref).lock();
```

Comment:
Just noting that we are reaching into an `Impl::` facility here so users won't technically be allowed to use it.

Reply:
Indeed. I guess this convoluted way of accessing the backend specialization stems from the original implementation strategy which I think aimed at providing some sort of *type erasure*.